services/blockstorage/adapter/memory/memory_persistence.go

```diff
@@ -119,7 +119,7 @@ func (bp *InMemoryBlockPersistence) GetBlockByTx(txHash primitives.Sha256, minBl
 	var candidateBlocks []*protocol.BlockPairContainer
 	for _, blockPair := range allBlocks {
 		bts := blockPair.TransactionsBlock.Header.Timestamp()
-		if maxBlockTs > bts && minBlockTs < bts {
+		if maxBlockTs >= bts && minBlockTs <= bts {
```

Comment:
this changes behavior - why?

Reply:
not really. these timestamps include grace periods. which mean they are of excessive range on purpose. the next paragraph goes over all the candidate blocks selected here and looks inside them exhaustively so if we add two more blocks to the candidate list it will not change the result, assuming the transaction is not found exactly at the outer edge of the grace period.

that is why the condition never broke any test in the past. but to be more strict in interpreting the spec i think these conditions should be inclusive. this is a minor fix.